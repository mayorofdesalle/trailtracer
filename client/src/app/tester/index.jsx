import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
