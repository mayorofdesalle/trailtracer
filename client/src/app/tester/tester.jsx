import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };