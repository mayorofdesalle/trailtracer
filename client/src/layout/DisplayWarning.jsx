import styled, { useTheme, keyframes } from 'styled-components';

import Container from '@components/ui/Container';
import Text, { Prim } from '@components/ui/Text';
import Icon from '@components/ui/Icon';

const orientate = (themeColors) => keyframes`
    0%, 25% {
        transform: rotate(-90deg);
        fill: ${themeColors.secondary};
    }

    50% {
        transform: rotate(0);
        fill: ${themeColors.primary};
    }
`;

const WarningContainer = styled(Container)`
    height: 100dvh;
    width: 100dvw;
    padding: 10%;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    background-color: ${({ theme }) => theme.colors.background};

    & > svg {
        --size: 100%;
        height: var(--size);
        width: var(--size);
        animation: ${({ theme }) => orientate(theme.colors)} 2s ease-in-out infinite alternate;
    }
`;

const DisplayWarning = () => {
    const theme = useTheme();
    return (
        <WarningContainer>
            <Icon name='phone-fill' color={theme.colors.primary} />
            <Text $heading $color={theme.colors.secondary} >CONTENT IS BEST DISPLAYED IN <Prim>PORTRAIT</Prim> ORIENTATION IN CURRENT RESOLUTION</Text>
        </WarningContainer>
    );
};

export default DisplayWarning;