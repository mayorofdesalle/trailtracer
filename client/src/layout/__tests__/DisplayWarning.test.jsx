import { describe, expect, it, render, screen } from '@app/tester';

import DisplayWarning from '../DisplayWarning';

describe('DisplayWarning', () => {

    it('should render correctly', () => {
        render(<DisplayWarning />);
        const warning = screen.getByRole('status');
        expect(warning).toBeInTheDocument();
    });

});