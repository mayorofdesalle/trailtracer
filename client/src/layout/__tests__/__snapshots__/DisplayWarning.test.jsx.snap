// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayWarning > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-kFCroH kHwwsn bkPqmz"
    role="status"
  >
    <svg
      class="sc-eDLKkx kBnHHP"
      fill="#435049"
      role="img"
    >
      <title>
        phone-fill
      </title>
      <use
        href="http://localhost:3000/src/assets/icons.svg#phone-fill"
      />
    </svg>
    <div
      class="sc-gLLuof dNuHMN"
    >
      displayWarning
    </div>
  </div>
</DocumentFragment>
`;
